ts/src/ChunkParser.ts

```diff
@@ -1,18 +1,25 @@
 import {Metadata} from "./grpc";
-import * as TextEncoding from "text-encoding";
 
 const HEADER_SIZE = 5;
 
-const global = Function('return this')();
+function decodeASCII(input: Uint8Array): string {
```

Comment:
We can pull this into a module, something like `metadata.ts`. With that, we can simplify the method name to be just `export function decode(input: Uint8Array) {...}`. We gain the ability to test the implementation.

With regards to correctness, the [spec](https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md) outlines that the possible values can be the following:
* Custom-Metadata → Binary-Header / ASCII-Header
* Binary-Header → {Header-Name "-bin" } {base64 encoded value}
* ASCII-Header → Header-Name ASCII-Value
* Header-Name → 1*( %x30-39 / %x61-7A / "_" / "-" / ".") ; 0-9 a-z _ - .
* ASCII-Value → 1*( %x20-%x7E ) ; space and printable ASCII

You're absolutely correct in suggesting that UTF-8 decoding is not necessary as it is not allowed by the spec.

In your implementation, we are missing the handling of all `printable` characters, that is anything between `0x9` through to `0xD`.
We can do something like this:
```javascript
// tab, newline, verticaltab, formfeed, carriage return
const isSpace = (char: number) => char >= 0x9 && char <= 0xD

function isValidGrpcAsciiValue(val: number): boolean {
    return isSpace(val) || (val >= 0x20 && val <= 0x7e)
}
```

Furthermore, I propose that instead of throwing an error (which we'll need to catch), we drop the invalid characters and log a warning for the engineers. Something like this:
```javascript
export function decode(input: Uint8Array): string {
  return input
    .filter(val => {
      const isValid = isValidGrpcAsciiValue(val);
      if (!isValid) {
        console.warn(`Encountered invalid character in metadata. Attempted to decode ${input} but ${val} is not a valid grpc metadata character. Dropping the character.`)
      }
      return isValid;
    })
    .map(val => ...)

}
```

Let me know what you think

Reply:
I think it's a good idea, but however it's not really easy to drop characters in ES5 (almost no extension methods on typed arrays).

As a result, I think bailing out with an error is the simplest solution here.

Reply:
Uh, and also, are characters other than CRLF allowed in HTTP 1.1 headers? I initially guessed it's not.
(The HTTP2 spec uses binary framing so none of those characters are used and thus not documented.)

Reply:
Looks like you're right, the HTTP 1.1 spec says 
```
3.2.3.  Whitespace

   This specification uses three rules to denote the use of linear
   whitespace: OWS (optional whitespace), RWS (required whitespace), and
   BWS ("bad" whitespace).

   The OWS rule is used where zero or more linear whitespace octets
   might appear.  For protocol elements where optional whitespace is
   preferred to improve readability, a sender SHOULD generate the
   optional whitespace as a single SP; otherwise, a sender SHOULD NOT
   generate optional whitespace except as needed to white out invalid or
   unwanted protocol elements during in-place message filtering.

   The RWS rule is used when at least one linear whitespace octet is
   required to separate field tokens.  A sender SHOULD generate RWS as a
   single SP.

   The BWS rule is used where the grammar allows optional whitespace
   only for historical reasons.  A sender MUST NOT generate BWS in
   messages.  A recipient MUST parse for such bad whitespace and remove
   it before interpreting the protocol element.

     OWS            = *( SP / HTAB )
                    ; optional whitespace
     RWS            = 1*( SP / HTAB )
                    ; required whitespace
     BWS            = OWS
                    ; "bad" whitespace
```
 so only `SP` or `HTAB` are allowed